Add bridge metrics (#3265)
strimzi/strimzi-kafka-operator: documentation/modules/metrics/kafka-bridge/proc_kafka-bridge-enabling-grafana.adoc

// Module included in the following assemblies:
//
// metrics/assembly_metrics-kafka-bridge.adoc

[id='proc-kafka-bridge-enabling-{context}']

> = Enabling the Kafka Bridge Grafana dashboard

Reviewer: Similar to the above ... do we really need this procedure? Can't we just keep the list of items from the dashabord where you describe the location and remove this?

Author: Same thought as before. Isn't it even enforced by PR #3269 with the refactoring made by Paul?

If you deployed Kafka Bridge with your Kafka cluster, you can enable Grafana to present the metrics data it exposes.

A Kafka Bridge dashboard is provided in the `examples/metrics` directory as a JSON file:

* `strimzi-kafka-bridge.json`

.Prerequisites

* Kafka Bridge is deployed with xref:proc-kafka-bridge-configuring-{context}[Kafka Bridge metrics configuration]
* xref:assembly-metrics-prometheus-{context}[Prometheus and Prometheus Alertmanager are deployed to the Kafka cluster]
* xref:proc-metrics-deploying-grafana-{context}[Grafana is deployed to the Kafka cluster]

This procedure assumes you already have access to the Grafana user interface and Prometheus has been added as a data source.
If you are accessing the user interface for the first time, see xref:assembly-metrics-grafana-{context}[Grafana].

.Procedure

> . Access the Grafana user interface.

Reviewer 2: Step as link to https://strimzi.io/docs/operators/master/deploying.html#proc-metrics-grafana-dashboard-str

Author: What do you mean?

Author: I don't see it as a link in the Kafka Exporter section for example.

> . Click *Dashboards*, then *Import* to open the _Import Dashboard_ window and import the example Kafka Bridge dashboard (or paste the JSON).

Reviewer 2: ```suggestion

```

Author: Why this change? In all the other places, I see the entire procedure to import a dashboard.

+
When metrics data has been collected for some time, the Kafka Bridge charts are populated.

> .Kafka Bridge Grafana charts

Reviewer 2: Format should be like other dashboards:

> Strimzi Kafka:: Shows metrics for:
> +
> * Brokers online count
> * Active controllers in the cluster count

From the metrics, you can create charts to display:

> * The number of open HTTP connections to the bridge

Reviewer 2: ```suggestion
* The number of open HTTP connections to the Kafka Bridge```

> * Requests being processed

Reviewer 2: ```suggestion
* HTTP requests being processed count
```

> * Requests processed/secs grouped by HTTP method

Reviewer 2: ```suggestion
* Requests processed per second grouped by HTTP method
```

> * The requests rate in total and even grouped by response codes (2XX, 4XX, 5XX)

Reviewer 2: ```suggestion
* The total request rate grouped by response codes (2XX, 4XX, 5XX)
```

> * Bytes received /sec and Bytes sent /sec

Reviewer 2: ```suggestion
* Bytes received and sent per second
```

> * Requests for each bridge endpoint

Reviewer 2: ```suggestion
* Requests for each Kafka Bridge endpoint
```

> * Number of Kafka consumers, producers, and related opened connections used by the bridge itself

Reviewer 2: ```suggestion
* Number of Kafka consumers, producers, and related opened connections used by the Kafka Bridge itself
```

* Kafka producer:
** The average number of records sent per second (grouped by topic)
** The number of outgoing bytes sent to all brokers per second (grouped by topic)
** The average number of records per second that resulted in errors (grouped by topic)
* Kafka consumer:
** The average number of records consumed per second (grouped by clientId-topic)
** The average number of bytes consumed per second (grouped by clientId-topic)
** Partitions assigned (grouped by clientId)

> * JVM related metrics

Reviewer 2: We listed the three for the other dashboards. Maybe this is better.
